src/pages/index.js

```diff
@@ -0,0 +1,323 @@
+const cardModalLinkInput = document.querySelector("#add-card-link-input");
+const cardModalCaptionInput = document.querySelector("#add-card-name-input");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const previewModalCloseButton = document.querySelector(
+  ".modal__close_type_preview"
+);
+const modalSubmitButton = cardModal.querySelector(
+  ".modal__submit-button_disabled"
+);
+const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
+const previewModalImageEl = document.querySelector(".modal__image");
+const previewModalCaptionEl = document.querySelector("#add-modal-caption");
+
+const profileAvatar = document.querySelector(".profile__avatar");
+
+//avatar form element
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarModalButton = document.querySelector(".profile__avatar-btn");
+const avatarModalCloseButton = avatarModal.querySelector(".modal__close");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__button");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+//delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalCloseButton = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+function getCardElement(data) {
+  let isLiked = data.isLiked;
+  const _id = data._id;
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardTitleEl = cardElement.querySelector(".card__title");
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardTitleEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+  cardLikeButton.addEventListener("click", (evt) => {
+    handleLike(evt);
+  });
+
+  if (isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  function handleLike(evt) {
+    api
+      .changeLikeStatus(_id, isLiked)
+      .then((data) => {
+        if (!isLiked) {
+          cardLikeButton.classList.add("card__like-button_liked");
+        } else {
+          cardLikeButton.classList.remove("card__like-button_liked");
+        }
+        isLiked = !isLiked;
+      })
+      .catch((err) => {
+        alert("Could not like/unlike card");
+        console.error(err);
+      });
+  }
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  cardImageEl.addEventListener("click", (evt) => {
+    handleImageClick(data);
+  });
+
+  deleteModalCloseButton.addEventListener("click", () => {
```

Comment:
* One remark hasn't been  corrected https://github.com/sasha-jovic/se_project_spots/pull/1#discussion_r1902200329  

Please, move `  deleteModalCloseButton.addEventListener("click", () => {` **outside** the `getCardElement` function into line 266